test/react_renderer_test.rb

```diff
@@ -2,6 +2,24 @@
 
 class ReactRendererTest < ActiveSupport::TestCase
 
+  test 'Rendering with a different json engine' do
+    require 'oj'
+    ::Oj.default_options = { :mode => :compat }
+
+    cfg = Rails.application.config.react
+    React::Renderer.setup!(cfg.react_js, cfg.components_js, {
+      :size => cfg.max_renderers, :timeout => cfg.timeout, :json_engine => ::Oj
+    })
+
+    result = React::Renderer.render "TodoList", :todos => %w{todo1 todo2 todo3}
+    assert_match(/todo1.*todo2.*todo3/, result)
+    assert_match(/data-react-checksum/, result)
```

Comment:
Is expecting the same rendered output as the test below? If so, how can I tell that `Oj` (not `JSON`) was used to create this output?

One way to be sure would be to make a dummy JSON library, for example 

```
class StupidJSON 
  def self.encode(*args)
    %{ { "some_key" : "some_value" } }
  end
end
```

Then use that dummy encoder and make sure you get the expected output.